src/ORM/DataQuery.php

```diff
@@ -1076,14 +1085,30 @@ protected function joinManyManyRelationship(
             if ($ancestorTable !== $componentBaseTable) {
                 $ancestorTableAliased = $componentPrefix . $ancestorTable;
                 $this->query->addLeftJoin(
-                    $ancestorTable,
+                    $this->getJoinTableName($ancestor, $ancestorTable),
                     "{$componentIDColumn} = \"{$ancestorTableAliased}\".\"ID\"",
                     $ancestorTableAliased
                 );
             }
         }
     }
 
+    /**
+     * Use this extension point to alter the table name
+     * useful for versioning for example
+     *
+     * @param $class
+     * @param $table
+     * @return mixed
+     */
+    protected function getJoinTableName($class, $table)
```

Comment:
would you be able to change this from a protected function to a private function.  If we keep it as protected then we'd need to release this as a new minor version (4.6), though if it's private we can release this as a patch (4.4.x)

Reply:
function changed to private